### PR TITLE
misc: Increase max failures in IATR badge to 99

### LINKED ISSUE #25662: Debug M1.1 - Update sidebar badge count display
**Current behavior**
The Debug page badge in the sidebar will show the failed test count up to 9 and then shows "9+" when the count is above 9.  

**Desired behavior**
The Debug page badge in the sidebar will show up to 99 and then show "99+" if the count is higher than 99.

* Show the actual count up to 99
* When above 99, show "99+"
* Offset the badge in the collapsed mode so that it has consistent right margin from the edge of the sidebar (see Figma).  

Figma: https://www.figma.com/file/PAOozuCxtfetqP8fUVvg0y/branch/QkCp2F69X0qyLo7qnYQ67e/Cypress-App?node-id=23899%3A229781&t=AeAPEJnkNCEES56e-1

### PR DESCRIPTION
**Pull request by mike-plummer**
- Closes #25662 

### Additional details
Changing layout of sidebar badge when collapsed - failing test count should now right-align to prevent overflow and maximum count is increased to 99.

The layout change should only impact the failure cases when count is `99+` - lower values, "0", and "New" states should retain current behavior of left-aligning and overflowing to the right when needed. This will be changing soon in follow-on work outside this ticket.

~**Note:** I have added a "min-width" condition for failure states so that the badge at least partially overlaps the debug icon when the number of failures is a single digit. Not having this gave it a somewhat disassociated look~

### Steps to test
Review `SidebarNavigation.cy.tsx`
Review Percy snapshots

### How has the user experience changed?
| State | Screenshot |
|------|-------------|
| Passing |  Screenshot 2023-02-07 at 9 47 06 AM|
| New | Screenshot 2023-02-07 at 9 42 02 AM |
| Failing - single digit |  Screenshot 2023-02-07 at 9 42 17 AM|
| Failing - double digit | Screenshot 2023-02-07 at 9 42 27 AM |
| Failing - > 100 | Screenshot 2023-02-07 at 9 42 35 AM |

### PR Tasks
- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/src/navigation/SidebarNavigationRow.vue [hunk 1]
```diff
@@ -82,7 +82,24 @@ const props = withDefaults(defineProps <{
...
+
+    // Right-align failure counts within sidebar (#25662)
+    if (props.badge.status === 'failed' || props.badge.status === 'error') {
+      // Add min-width so that single-digit badge overlaps sidebar icon at least a little
+      classes.push('right-4px', 'min-w-20px', 'text-center')
+    } else {
+      // Non-failures should left-align and overflow sidebar if needed
+      classes.push('left-36px')
+    }
+  }
```
Comment (warrensplayer) [inline packages/app/src/navigation/SidebarNavigationRow.vue]: I interpreted Emil's comment in Slack to mean that the right alignment should only apply when there are three characters to display for the error count in the badge in the collapsed state (i.e. it is showing "99+"). The scenarios where it only shows 1 or 2 digits can stay in their original positions and not need the min width and would not extend past the collapsed sidebar's right border.

For this update, I think you could get away with only right aligning the badge if the value is `99+`, but you could also be a bit more generic and check the status as well as the length of the value that will show in the badge.

Comment (mike-plummer) [reply packages/app/src/navigation/SidebarNavigationRow.vue]: Hmm, I see what you mean - I glossed over that last bullet point. This disagrees with the [original Figma](https://www.figma.com/file/PAOozuCxtfetqP8fUVvg0y/branch/QkCp2F69X0qyLo7qnYQ67e/Cypress-App?node-id=23899%3A229781&t=AeAPEJnkNCEES56e-1) for this which showed right-alignment for an `80` and `99+` case (but no single-digit case, hence this confusion).

Comment (mike-plummer) [reply packages/app/src/navigation/SidebarNavigationRow.vue]: Updated in 8fb6cb92dc11c073ae3a249e29f4ce9b79f20425

**Review by warrensplayer — APPROVED** [review]
Looks good!

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**